popsycle/synthetic.py

```diff
@@ -604,6 +604,24 @@ def perform_pop_syn(ebf_file, output_root, iso_dir,
             num_stars_in_bin = 2e6
             num_bins = int(math.ceil(len_adx / num_stars_in_bin))
 
+            # Create the KDTree used for calculating extinction from the first
+            # sample of stars
+            kdt_star_p = None
+            kdt_star_exbv = None
+            if len_adx > 0:
+                num_kdtree_samples = int(min(len_adx, 2e6))
```

Comment:
The current version has 2e6 hard coded into it. I believe @caseylam chose this value because it was the most number of stars that could be loaded in without crashing memory. I'm not sure I understand the rest of your comment. Why would the choice of the size of the bin change the diversity or repetition of the compact objects in the data?

Reply:
comments added